lib/offline/OfflineStorage_Room.cpp

```diff
@@ -70,52 +70,57 @@ namespace MAT_NS_BEGIN
     }
 
     /**
-     * call PushLocalFrame and track current depth
-     */
+             * call PushLocalFrame and track current depth
+             */
 
     void
     OfflineStorage_Room::ConnectedEnv::pushLocalFrame(uint32_t frameSize)
     {
-        try {
-            if (env->PushLocalFrame(frameSize) == JNI_OK) {
-                ++push_count;
-            }
+        if (env->PushLocalFrame(frameSize) == JNI_OK)
+        {
+            ++push_count;
         }
-        catch (std::exception e) {
-            auto what = e.what();
-            if (!what) {
-                what = "*nothing*";
+        if (env->ExceptionCheck() == JNI_TRUE)
+        {
+            env->ExceptionDescribe();
+            env->ExceptionClear();
+            if (s_throwExceptions)
+            {
+                MATSDK_THROW(std::runtime_error("Push Local Frame"));
```

Comment:
please check if this will lead to a crash

Reply:
The exception, if thrown, is caught on line 306.